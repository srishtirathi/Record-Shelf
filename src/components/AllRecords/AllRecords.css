.app-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    text-align: center;
    flex-wrap: wrap;
}

.inventory-container:nth-child(2n+1) {
    background: rgb(223, 223, 223)
}

.inventory-container:nth-child(2n) {
    background: rgb(171, 171, 171)
}

.allcategory-button {
    background: url(/src/assets/icon-genre.svg) no-repeat;
    cursor: pointer;
    border: none;
    padding: 2rem
}

.inventory-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    padding: 2rem;
}

@media screen and (max-width: 27rem) {
    .container {
        flex-direction: column;
        justify-content: center;
    }
}