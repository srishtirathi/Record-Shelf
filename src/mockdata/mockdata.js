/* eslint-disable import/prefer-default-export */
export const mockAllRecords = [
  {
    id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
    name: 'Aankh Marey (From "Simmba")',
    genre: {
      id: '37146752-2c95-4204-a845-a32f3cdce9c8',
      name: 'Bollywood',
    },
    artist: {
      id: 'b22d7ed5-3ded-4958-a036-60453b7fc102',
      name: 'Neha Kakkar',
    },
    albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
    publishedAt: '2018-12-06T00:00:00',
  }];

export const mockrecordByCategory = [{
  Bollywood: {
    id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
    name: 'Aankh Marey (From "Simmba")',
    genre: {
      id: '37146752-2c95-4204-a845-a32f3cdce9c8',
      name: 'Bollywood',
    },
    artist: {
      id: 'b22d7ed5-3ded-4958-a036-60453b7fc102',
      name: 'Neha Kakkar',
    },
    albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
    publishedAt: '2018-12-06T00:00:00',
  },
}];
