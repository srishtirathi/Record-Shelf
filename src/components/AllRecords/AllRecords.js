/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getRecords } from '../../utils/api';
import Card from '../Card/Card';
import './AllRecords.css';
import { groupByCategory } from '../../utils/recordUtil';
import RecordsByCategory from '../RecordsByCategory/RecordsByCategory';

const AllRecords = ({
  allRecords, increaseCount, decreaseCount, updateLike,
}) => (

  <div>
    <div>
      <Link to="/category"><button type="button" className="allcategory-button" /></Link>
    </div>
    <br />
    <h1>all songs</h1>
    <div className="app-container">
      {/*
      {allRecords.map((record) => (<h1>{record.name}</h1>))}

      {
        allRecords.map((record) => (
          <div>
            <h1>{record.name}</h1>

          </div>
        ))
      } */}

      {/* {allRecords.map((record) => (<Card record={record} />))} */}

      {allRecords.map((record) => (
        <div className="inventory-container">
          <Card
            name={record.name}
            artist={record.artist.name}
            albumArtUrl={record.albumArtUrl}
            count={record.count}
            like={record.like}
            increaseCount={increaseCount}
            record={record}
            decreaseCount={decreaseCount}
            updateLike={updateLike}
          />
        </div>
      ))}
    </div>
  </div>
);
export default AllRecords;
