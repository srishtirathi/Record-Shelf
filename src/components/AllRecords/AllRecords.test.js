/* eslint-disable react/jsx-props-no-spreading */
import { render } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import AllRecords from './AllRecords';
import { mockAllRecords } from '../../mockdata/mockdata';

describe(AllRecords.name, () => {
  const mockProps = {
    allRecords: mockAllRecords,
    map: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><AllRecords {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
});
