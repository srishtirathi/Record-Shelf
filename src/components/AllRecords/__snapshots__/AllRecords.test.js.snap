// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllRecords should match snapshot 1`] = `
<div>
  <div
    class="allrecords"
  >
    <div>
      <a
        href="/category"
      >
        <button
          class="category-button"
          type="button"
        >
          category
        </button>
      </a>
    </div>
    <br />
    <div>
      <div
        class="record-card"
        data-testid="card"
      >
        <img
          alt="songImg"
          class="song-img"
          data-testid="record-img"
          src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
        />
        <div
          class="record-container"
        >
          <p>
            Aankh Marey (From "Simmba")
          </p>
          <p>
            Neha Kakkar
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
